pilot/pkg/model/sidecar_test.go

```diff
@@ -1888,34 +2009,38 @@ func TestCreateSidecarScope(t *testing.T) {
 			}
 
 			for _, s1 := range sidecarScope.services {
-				found = false
+				serviceFound = false
+				portsMatched = false
+				var ports PortList
 				for _, s2 := range tt.excpectedServices {
 					if s1.Hostname == s2.Hostname {
+						serviceFound = true
 						if len(s2.Ports) > 0 {
 							if reflect.DeepEqual(s2.Ports, s1.Ports) {
-								found = true
-								break
+								portsMatched = true
+							} else {
+								ports = s2.Ports
 							}
-						} else {
-							found = true
-							break
 						}
+						break
 					}
 				}
-				if !found {
+				if !serviceFound {
 					t.Errorf("Expected service %v in SidecarScope but not found", s1.Hostname)
+				} else if len(ports) > 0 && !portsMatched {
```

Comment:
If I read the code correctly you don't actually need `ports`.
The if condition only checks `!portsMatched` and the error's got ports takes `s2.Ports`

Reply:
I think it is required because s2 is accessible only inside the inner loop.

Reply:
When ports are assigned, we break out of the loop and that is what we want to validate i.e. the service is matched and whether ports have been matced or not